q2_diversity/plugin_setup.py

```diff
@@ -240,6 +240,77 @@
     citations=[citations['legendrelegendre']]
 )
 
+plugin.methods.register_function(
+    function=q2_diversity.tsne,
+    inputs={'distance_matrix': DistanceMatrix},
+    parameters={
+        'number_of_dimensions': Int % Range(1, None)
+    },
+    outputs=[('tsne', PCoAResults)],
+    input_descriptions={
+        'distance_matrix': ('The distance matrix on which t-SNE should be '
+                            'computed.')
+    },
+    parameter_descriptions={
+        'number_of_dimensions': "Dimensions to reduce the distance matrix to. "
+                                "This number determines how many "
+                                "eigenvectors and eigenvalues are returned,"
+                                "and influences the choice of algorithm used "
+                                "to compute them. "
+                                "By default, uses the default "
+                                "eigendecomposition method, SciPy's eigh, "
+                                "which computes all eigenvectors "
+                                "and eigenvalues in an exact manner. For very "
+                                "large matrices, this is expected to be slow. "
+                                "If a value is specified for this parameter, "
+                                "then the fast, heuristic "
+                                "eigendecomposition algorithm fsvd "
+                                "is used, which only computes and returns the "
+                                "number of dimensions specified, but suffers "
+                                "some degree of accuracy loss, the magnitude "
+                                "of which varies across different datasets."
+    },
+    output_descriptions={'tsne': 'The resulting t-SNE matrix.'},
+    name='t-distributed stochastic neighbor embedding',
+    description=("Apply t-distributed stochastic neighbor embedding."),
+
+)
+
+plugin.methods.register_function(
+    function=q2_diversity.uMAP,
```

Comment:
I think it would be more consistent to use all lowercase character for the function name, i.e. umap instead of uMAP. Same applies for the parameter regarding the output artifact

Reply:
but beware of name clashed between sklearn umap an qiime umap

Reply:
"but beware of name clashed between sklearn umap an qiime umap" 

Thats why I used mixed cases but I'll rename the import to up so umap is available for the function